Sources/IRGen/Component/IRAssignment.swift

```diff
@@ -5,30 +5,30 @@
 //  Created by Hails, Daniel R on 29/08/2018.
 //
 import AST
+import YUL
 
 /// Generates code for an assignment.
 struct IRAssignment {
-  var lhs: Expression
-  var rhs: Expression
+  var lhs: AST.Expression
+  var rhs: AST.Expression
 
-  func rendered(functionContext: FunctionContext, asTypeProperty: Bool = false) -> ExpressionFragment {
+  func rendered(functionContext: FunctionContext, asTypeProperty: Bool = false) -> YUL.Expression {
     let rhsIr = IRExpression(expression: rhs).rendered(functionContext: functionContext)
-    let rhsCode = rhsIr.expression
+    let rhsCode = rhsIr.description
 
     switch lhs {
     case .variableDeclaration(let variableDeclaration):
       let mangledName = Mangler.mangleName(variableDeclaration.identifier.name)
```

Comment:
alright — this was a cosmetic thing which could've been nice. We could add this convenience constructor as an extension so that it only affects this module (means that we don't have to deal with the circular dependencies). But it lessens the benefits.